src/app/Shared/Context/DataContext.tsx

```diff
@@ -14,7 +13,7 @@ const defaultProfile: DataInterface = {
     ZNData: [],
     ZNDataFiltered: [],
     keyValues: [],
-    uniqueValues: defaultFilter,
+    uniqueValues: {},
```

Comment:
See FilterContext: Try to use null for expressing empty state

Reply:
if I use "undefined" as empty state, I have to check in every function "if (x !== undefined) {..} else {retrun ""}" or something like that. What would be the advantage of this? (esLint says that I should use undefined instead od null https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v20.1.0/docs/rules/no-null.md)

Reply:
I would disable this linter rule. null and undefined have different semantics. E.g. take the following api response type
user: {
id: string
name: string | null
address?: string
}
Here address is optional. So it is undefined or string. If it is undefined the data was not send. But it might be present on the server.
Name is string | null. So name = null states the data is there but it is explicitly set to null. So the user has no defined name.
So undefined is like optional and null like explicitly set to null.

So i don't know the semantics of your code here. I think uniqueValues is not optional. Perhaps the defaultFilter was the best solution. Nevertheless if you use {} as a value the compiler does not complain about that but your code can still crash if you not check if the properties of uniqueValues are set. In this case checking for null is the easier solution.